src/lib/TypeExtensions/OpenApiDocumentExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.OpenApi.ApiManifest.Helpers;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.ApiManifest.TypeExtensions
+{
+    public static class OpenApiDocumentExtensions
+    {
+        private const string DefaultPublisherName = "publisher-name";
+        private const string DefaultPublisherEmail = "publisher-email@example.com";
+
+        /// <summary>
+        /// Converts an <see cref="OpenApiDocument"/> to an <see cref="ApiManifestDocument"/>.
+        /// </summary>
+        /// <param name="document">The OpenAPI document to convert.</param>
+        /// <param name="apiDescriptionUrl">The URL of the API description.</param>
+        /// <param name="applicationName">The name of the application.</param>
+        /// <param name="apiDependencyName">The name of the API dependency.</param>
+        /// <returns>An <see cref="ApiManifestDocument"/>.</returns>
+        public static ApiManifestDocument ToApiManifest(this OpenApiDocument document, string? apiDescriptionUrl, string applicationName, string? apiDependencyName = default)
+        {
+            ArgumentNullException.ThrowIfNull(document);
+            ValidationHelpers.ValidateNullOrWhitespace(nameof(apiDescriptionUrl), apiDescriptionUrl, nameof(ApiManifestDocument));
+            ValidationHelpers.ValidateNullOrWhitespace(nameof(applicationName), applicationName, nameof(ApiManifestDocument));
+
+            apiDependencyName = NormalizeApiName(apiDependencyName ?? document.Info.Title);
+            var publisherName = document.Info.Contact?.Name ?? DefaultPublisherName;
+            var publisherEmail = document.Info.Contact?.Email ?? DefaultPublisherEmail;
```

Comment:
I think we should have an option to override publisher information with parameters, as a publisher may not necessarily be the same person that owns the OpenAPI document.